src/controller/c/supervisor.c

```diff
@@ -949,12 +1037,32 @@ static void supervisor_read_answer(WbDevice *d, WbRequest *r) {
         node_id = uid;
       }
     } break;
+    case C_SUPERVISOR_NODE_GET_PROTO: {
+      const int id = request_read_int32(r);
+      const bool is_derived = request_read_uchar(r) == 1;
+      const int num_fields = request_read_int32(r);
+      const char *type_name = request_read_string(r);
+      if (id < 0)
+        break;
+      WbProtoInfoStruct *p = malloc(sizeof(WbProtoInfoStruct));
```

Comment:
Isn't this a memory leak? Specifically, doesn't this result in a new `WbProtoInfoStruct` being allocated every time `wb_supervisor_node_get_proto` is called?

Reply:
No, if `wb_supervisor_node_get_proto` has already been called, [the proto information will be cached in `node-proto_info`](https://github.com/cyberbotics/webots/blob/56a6fcf68bd6a94ba895000f9f6f06b51b92f935/src/controller/c/supervisor.c#L2928), and no request will be sent to Webots. The only exception is if the node is regenerated, in which case [the old proto structs are freed](https://github.com/cyberbotics/webots/blob/56a6fcf68bd6a94ba895000f9f6f06b51b92f935/src/controller/c/supervisor.c#L417https://github.com/cyberbotics/webots/blob/56a6fcf68bd6a94ba895000f9f6f06b51b92f935/src/controller/c/supervisor.c#L417).

Reply:
Nevermind. I see now that this is handled in `wb_supervisor_node_get_proto`.